### ISSUE TITLE
[Bug]: Don't serve the artist fallback image over the API, but send a 404

### ISSUE THREAD

**Issue by tzugen** [navidrome/navidrome#2575]
### I confirm that:

- [X] I have searched the existing [open AND closed issues](https://github.com/navidrome/navidrome/issues?q=is%3Aissue) to see if an issue already exists for the bug I've encountered
- [X] I'm using the latest version (your issue may have been fixed already)

### Version

latest

### Current Behavior

Hi, i am tzugen, one of the main devs of the Android Ultrasonic app.

A navidrome user opened[ this issue with](https://gitlab.com/ultrasonic/ultrasonic/-/issues/1268) us.
They changed the artist cover art, and it didn't update in our app.
That we don't have a button to clear the image cache is on us,
but I think its not the best behaviour that navidrome sends a fallback image over the network if there is no actual cover art.

In our app, we have a high quality fallback image that takes its colours dynamically from the current theme and colors.
Obviously we cannot easily detect whether the image we receive over the wire is actually a real image or a raster fallback. 

### Expected Behavior

I would suggest that if a cover image is requested that navidrome doesnt have it simply responds with a 404 or similar, so that clients can display the fallback image of their choosing.

### Code of Conduct

- [X] I agree to follow Navidrome's Code of Conduct

**Comment by deluan** [comment]
Hey @tzugen, thanks for reaching out. I had this request from another Subsonic client dev, but I'm just concerned it would break other clients. I'll do some testing with a few clients and see if this can be changed.

**Comment by deluan** [comment]
Doing some tests with other clients and the OG Subsonic itself, I see the `getCoverArt` endpoint also returns a placeholder if a image is not found. 

Example:
http://demo.subsonic.org/rest/getCoverArt.view?u=guest&p=guest&f=json&v=1.6.0&c=Test&id=9999998

Results:
![Screenshot 2024-02-05 at 8 36 09 PM](https://github.com/navidrome/navidrome/assets/331353/0928ad41-a198-4a77-992e-3fa097774ca8)

I understand the reasoning for asking for this, and I plan to implement it (at least as an option), but I'd like to understand from @tzugen and other client devs that asks for this: How do you handle this for other servers like Subsonic itself (and probably all its forks: Airsonc, Madsonic...)

**Comment by dweymouth** [comment]
How I handle it in Supersonic, is I have my own placeholder that adapts to the theming of the app (light/dark mode, and user-supplied color customizations). You can see how it looks for artists from the screenshots in this LMS feature request to add the artist images: https://github.com/epoupon/lms/issues/392

The same placeholder is also shown while waiting for an image to be fetched, if it's not yet in Supersonic's cover cache.

I don't make any attempt to replace Navidrome's placeholder images with my own, so in Navidrome's case, the supplied placeholders will be shown, but Supersonic's placeholder is used when waiting for the first cover fetch, as described. I think it's better for apps to be able to use their own placeholders, which may fit their styling better, and it is difficult/not worth it to try to filter out the placeholder image supplied by Navidrome/OG Subsonic to replace it.

**Comment by deluan** [comment]
> I think it's better for apps to be able to use their own placeholders, which may fit their styling better, and it is difficult/not worth it to try to filter out the placeholder image supplied by Navidrome/OG Subsonic to replace it.

I totally agree, but I tried to replicate OG Subsonic behaviour as close as possible to avoid breaking any old (and possible unmaintained) clients. I may introduce a config option to control this behaviour.

**Comment by epoupon** [comment]
Hello! I just found this as it was linked to an issue on my repo. For LMS, there is a configuration option to list the Subsonic clients that are served default covers:

```
# List of clients for whom a default cover is served (as they do not have their own)
api-subsonic-default-cover-clients = ("DSub", "Substreamer");
```

Clients not on the list will receive a 404 error.
I have not re-checked the list, but at the time I added the option, that's all I found.
